lib/apimatic-core/utilities/logger_helper.rb

```diff
@@ -0,0 +1,67 @@
+module CoreLibrary
+  # logger helper methods.
+  class LoggerHelper < ApiLogger
+    def self.get_content_type(headers)
+      return '' if headers.nil?
+
+      headers.find { |key, _| key.downcase == 'content-type' }&.last || ''
+    end
+
+    def self.get_content_length(headers)
+      return '' if headers.nil?
+
+      headers.find { |key, _| key.downcase == 'content-length' }&.last || ''
+    end
+
+    def self.extract_headers_to_log(headers_to_include, headers_to_exclude, headers_to_unmask, headers)
+      return headers if headers.nil?
+
+      filtered_headers = {}
+
+      if headers_to_include.any?
+        headers_to_include.each do |name|
+          key = headers.keys.find { |header_key| header_key.downcase == name.downcase }
+          filtered_headers[key] = headers[key] if headers[key]
+        end
+      elsif headers_to_exclude.any?
+        headers.each do |key, val|
+          filtered_headers[key] = val unless headers_to_exclude.any? do |excluded_name|
+                                               excluded_name.downcase == key.downcase
+                                             end
+        end
+      else
+        filtered_headers = headers
+      end
+
+      mask_sensitive_headers(filtered_headers, headers_to_unmask)
+    end
+
+    def self.mask_sensitive_headers(headers, headers_to_unmask)
+      return headers unless @mask_sensitive_headers || !headers.nil?
+
+      headers.each do |key, val|
+        headers[key] = mask_if_sensitive_header(key, val, headers_to_unmask)
+      end
+
+      headers
+    end
+
+    def self.mask_if_sensitive_header(name, value, headers_to_unmask)
+      non_sensitive_headers = %w[
```

Comment:
Nits, Non-blocker: instead of initializing the list with each call to the method, you can make it a constant list outside of the method and access it in the method and also apply the downcase operation only 1 time on the list.